lib/db/models/SwapDeal.ts

```diff
@@ -0,0 +1,39 @@
+import Sequelize from 'sequelize';
+import { db } from '../../types';
+
+export default (sequelize: Sequelize.Sequelize, DataTypes: Sequelize.DataTypes) => {
+  const attributes: db.SequelizeAttributes<db.SwapDealAttributes> = {
+    role: { type: DataTypes.TINYINT, allowNull: false },
+    state: { type: DataTypes.TINYINT, allowNull: false },
+    phase: { type: DataTypes.TINYINT, allowNull: false },
+    errorReason: { type: DataTypes.STRING, allowNull: true },
+    r_hash: { type: DataTypes.STRING, allowNull: false, unique: true },
+    r_preimage: { type: DataTypes.STRING, allowNull: true },
+    peerPubKey: { type: DataTypes.STRING, allowNull: false },
+    orderId: { type: DataTypes.STRING, allowNull: false },
+    localId: { type: DataTypes.STRING, allowNull: false },
+    proposedQuantity: { type: DataTypes.DECIMAL(8), allowNull: false },
+    quantity: { type: DataTypes.DECIMAL(8), allowNull: true },
+    pairId: { type: DataTypes.STRING, allowNull: false },
+    takerAmount: { type: DataTypes.NUMBER , allowNull: false },
```

Comment:
There is no Data Type called `NUMBER` in sequelize, which is why the tests are failing. `takerAmount` should be a `BIGINT`.